plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetsInstallationTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.tools.eclipse.util.FacetedProjectHelper;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.zip.ZipFile;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.util.internal.ZipUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+/**
+ * Tests some cases of installing the App Engine Standard facet on existing projects.
+ */
+public class StandardFacetsInstallationTests {
+  private IWorkspaceRoot root;
+  private IProject project;
+
+  @Before
+  public void setUp() {
+    root = ResourcesPlugin.getWorkspace().getRoot();
+  }
+
+  @After
+  public void tearDown() {
+    if (project != null) {
+      try {
+        project.delete(true, null);
+      } catch (CoreException ex) {
+        ex.printStackTrace();
+      }
+      assertFalse("project should have been deleted", project.exists());
+    }
+  }
+
+  @Test
+  public void testStandardFacetInstallationOnGwtWar() throws IOException, CoreException {
+    project = importProject("projects/gwt-project.zip");
```

Comment:
It was created using the _New > GWT Application wizard_.  I then added the _Dynamic Web Module_ 2.5.

I can re-create it from scratch if needed.